pandas/core/internals/blocks.py

```diff
@@ -991,6 +991,18 @@ def setitem(self, indexer, value):
 
         # set
         else:
+            if (
```

Comment:
elif?

Reply:
no, we still want to fall through to L1006

Reply:
sure but i thin it is much more clear to use an elif and then simply repeat that line in the else

Reply:
updated + green